bindings/cs/MongoDB.Libmongocrypt/KmsCredentials.cs

```diff
@@ -19,40 +19,32 @@
 namespace MongoDB.Libmongocrypt
 {
     /// <summary>
-    /// Local KMS Credentials.
+    /// KMS Credentials.
     /// </summary>
-    /// <seealso cref="IKmsCredentials" />
-    /// <seealso cref="IInternalKmsCredentials" />
-    public class LocalKmsCredentials : IKmsCredentials, IInternalKmsCredentials
+    public class KmsCredentials
     {
+        private readonly byte[] _credentialsBytes;
+
         /// <summary>
-        /// Creates an <see cref="LocalKmsCredentials"/> class.
+        /// Creates an <see cref="KmsCredentials"/> class.
         /// </summary>
-        /// <param name="key">The key.</param>
-        public LocalKmsCredentials(byte[] key)
+        /// <param name="credentialsBytes">The bytes representation of credentials bson document.</param>
+        public KmsCredentials(byte[] credentialsBytes)
         {
-            Key = key;
+            _credentialsBytes = credentialsBytes ?? throw new ArgumentNullException(nameof(credentialsBytes));
         }
-        public KmsType KmsType => KmsType.Local;
-
-        /// <summary>
-        /// Gets or sets the key.
-        /// </summary>
-        /// <value>
-        /// The key.
-        /// </value>
-        public byte[] Key { get; }
 
-        void IInternalKmsCredentials.SetCredentials(MongoCryptSafeHandle handle, Status status)
+        // internal methods
+        internal void SetCredentials(MongoCryptSafeHandle handle, Status status)
         {
             unsafe
             {
-                fixed (byte* p = Key)
+                fixed (byte* p = _credentialsBytes)
                 {
                     IntPtr ptr = (IntPtr)p;
-                    using (PinnedBinary pinned = new PinnedBinary(ptr, (uint)Key.Length))
+                    using (PinnedBinary pinned = new PinnedBinary(ptr, (uint)_credentialsBytes.Length))
                     {
-                        handle.Check(status, Library.mongocrypt_setopt_kms_provider_local(handle, pinned.Handle));
+                        handle.Check(status, Library.mongocrypt_setopt_kms_providers(handle, pinned.Handle));
```

Comment:
I don't fully understand the functionality of native library, so @DmitryLukyanov I rely on your changes regarding the library function calls.

Reply:
the only logic here is passing bytes into `mongocrypt_setopt_kms_providers` method which is:

      bool
      mongocrypt_setopt_kms_providers (mongocrypt_t *crypt,
                                       mongocrypt_binary_t *kms_providers)

fully the same as it was before:

     bool
     mongocrypt_setopt_kms_provider_local (mongocrypt_t *crypt,
                                      mongocrypt_binary_t *key)